hbase-client/src/main/java/org/apache/hadoop/hbase/CheckAndMutate.java

```diff
@@ -0,0 +1,312 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.hadoop.hbase;
+
+import org.apache.hadoop.hbase.client.Delete;
+import org.apache.hadoop.hbase.client.Put;
+import org.apache.hadoop.hbase.client.Row;
+import org.apache.hadoop.hbase.client.RowMutations;
+import org.apache.hadoop.hbase.filter.Filter;
+import org.apache.hadoop.hbase.io.TimeRange;
+import org.apache.hadoop.hbase.util.Bytes;
+import org.apache.hbase.thirdparty.com.google.common.base.Preconditions;
+import org.apache.yetus.audience.InterfaceAudience;
+
+
+/**
+ * Used to perform batch CheckAndMutate operations.
+ *
+ * Use the builder classes to instantiate a CheckAndMutate object.
+ * This builder classes are fluent style APIs, the code are like:
+ *
+ * <pre>
+ * <code>
+ * CheckAndMutate checkAndMutate = CheckAndMutate.builder(row, family).qualifier(qualifier)
+ *   .ifNotExists().build(put);
+ * </code>
+ * <code>
+ * CheckAndMutate checkAndMutate = CheckAndMutate.builder(row, filter).build(put);
+ * </code>
+ * </pre>
+ */
+@InterfaceAudience.Public
+public class CheckAndMutate implements Row {
+
+  /**
+   * A builder class for building a CheckAndMutate object.
+   */
+  public static class Builder {
+    private final byte[] row;
+    private final byte[] family;
+    private byte[] qualifier;
+    private CompareOperator op;
+    private byte[] value;
+    private TimeRange timeRange;
+
+    private Builder(byte[] row, byte[] family) {
+      this.row = Preconditions.checkNotNull(row, "row is null");
+      this.family = Preconditions.checkNotNull(family, "family is null");
+    }
+
+    /**
+     * @param qualifier column qualifier to check
+     * @return the builder object
+     */
+    public Builder qualifier(byte[] qualifier) {
+      this.qualifier = Preconditions.checkNotNull(qualifier, "qualifier is null. Consider using" +
+        " an empty byte array, or just do not call this method if you want a null qualifier");
+      return this;
+    }
+
+    /**
+     * @param timeRange timeRange to check
+     * @return the builder object
+     */
+    public Builder timeRange(TimeRange timeRange) {
+      this.timeRange = timeRange;
+      return this;
+    }
+
+    /**
+     * Check for lack of column
+     * @return the builder object
+     */
+    public Builder ifNotExists() {
+      this.op = CompareOperator.EQUAL;
+      this.value = null;
+      return this;
+    }
+
+    /**
+     * Check for equality
+     * @param value the expected value
+     * @return the builder object
+     */
+    public Builder ifEquals(byte[] value) {
+      return ifMatches(CompareOperator.EQUAL, value);
+    }
+
+    /**
+     * @param compareOp comparison operator to use
+     * @param value the expected value
+     * @return the builder object
+     */
+    public Builder ifMatches(CompareOperator compareOp, byte[] value) {
```

Comment:
s/ifMatches/ifCompares/?

Reply:
We already have the same name as this method in Table$CheckAndMutateBuilder and AsyncTable$CheckAndMutateBuilder, so I named this method as the same name.

Table$CheckAndMutateBuilder#ifMatches:
https://github.com/brfrn169/hbase/blob/f66cbe1a40cd9d4cd4abd27c7ac9d1a5168b885f/hbase-client/src/main/java/org/apache/hadoop/hbase/client/Table.java#L346

AsyncTable$CheckAndMutateBuilder#ifMatches:
https://github.com/brfrn169/hbase/blob/f66cbe1a40cd9d4cd4abd27c7ac9d1a5168b885f/hbase-client/src/main/java/org/apache/hadoop/hbase/client/AsyncTable.java#L269